cockatrice/src/gamesmodel.cpp

```diff
@@ -472,10 +461,10 @@ bool GamesProxyModel::filterAcceptsRow(int sourceRow) const
     if ((int)game.max_players() > maxPlayersFilterMax)
         return false;
 
-    if (maxGameAgeSeconds != DEFAULT_MAX_GAME_AGE_SECONDS) {
-        QDateTime then;
-        then.setTime_t(game.start_time());
-        if (then.secsTo(QDateTime::currentDateTime()) > maxGameAgeSeconds) {
+    if (maxGameAge.isValid()) {
+        QDateTime now = QDateTime::currentDateTimeUtc();
+        QDateTime total = now.addSecs(-(qint64)game.start_time()); // cast to 64 bit value before 2038-1-19 comes around
```

Comment:
in 2038 this value will become 1<<31 and wrap to negative if it were cast to a plain int

Reply:
Lets be more careful about wrapping then. I'd consider changing the name of the variable then to indicate casting is a bad idea

Reply:
if you look at the if statement above you'll we we should consider never allowing more than 2147483647 players